quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -235,7 +238,15 @@ public void validateConfig(List<String> cliArgs, AbstractCommand abstractCommand
             checkChangesInBuildOptions((key, oldValue, newValue) -> {
                 if (key.startsWith(KC_PROVIDER_FILE_PREFIX)) {
                     if (timestampChanged(oldValue, newValue)) {
-                        throw new PropertyException("A provider JAR was updated since the last build, please rebuild for this to be fully utilized.");
+                        if (warnedTimestampChanged
+                                || Configuration.getOptionalBooleanKcValue("run-in-container").orElse(false)) {
```

Comment:
Just a nitpick for better readability, do we need the condition for `warnedTimestampChanged`?
```suggestion
                        if (Configuration.getOptionalBooleanKcValue("run-in-container").orElse(false)) {
```

Reply:
I just using the that prevent additional configuration calls as if 1 provider fails they likely all will. It's just a small optimization, so if you don't like it, it can go.